### DISCUSSION TITLE
Projects (beta) - Show sum of "number" field(s)

### DISCUSSION THREAD

**Discussion by petewalker** [community/community#5376] [Projects and Issues] [answered]
We're likely to use the number type field for sizing issues (as I believe is probably the most common enterprise use-case). There's no obvious way in the UI I could see to sum the total of a number field.

It would be useful to see:

* The sum of a particular number field at the top of a column in the board view - so you could quickly see the total story points in that column
  e.g. 
  <img width="362" alt="Screenshot 2021-08-31 at 12 18 50" src="https://user-images.githubusercontent.com/453816/131493958-4f76b9f5-f60c-4af8-9f0b-5e760a379280.png">

* The sum of a number field at the top of the group
* In table view, the ability to select multiple size cells, and see the total of those cells as you drag - like you can in excel or google sheets
  e.g.
  <img width="189" alt="Screenshot 2021-08-31 at 12 18 05" src="https://user-images.githubusercontent.com/453816/131493595-9e654df6-ba29-4a15-9e6e-e4f4141a41c1.png">

Our main use cases for this would be to:

* See total story points done vs. in backlog
* Experiment with different combinations of issues to see what we can fit into future sprints vs. velocity

**Comment by felixfbecker** [comment]
This is my number one feature request for the new issue tables, an absolute must-have for using this for iteration planning. 🥇 

It would be important here that the sum is shown only for the rows visible in the filtered view and **per-group**, so it is possible to e.g. see the sum of points/estimates for "TODO" items independent of "done" and "in progress" items. If/when [grouping by assignee](https://github.com/github/feedback/discussions/5448) is supported in the future, that would also be important so workload can be seen per team member and nobody is overloaded.

**Reply by ghost** [reply]
Yup this is the missing link to do proper planification based on estimate with github project. It's strange that it wasn't planned in the first place

**Comment by rodsfreitas** [comment]
Totally agree. Being able to sum up numeric fields would help a lot when planning the sprint.

**Comment by nathancolgate** [comment]
We wrote a very simple chrome extension that sums up float values in a column labeled "Estimate (hours)" in GitHub's new project beta UI. It alerts the total to the user via javascript.

It is very crude, but we use it _so_ much that I thought it would be worth sharing here:

https://github.com/brandnewbox/Github-Project-Column-Total-Chrome-Extension

**Comment by thenonameguy** [comment]
Thhanks @nathancolgate! 

I made a bookmarklet of your script using: https://mrcoles.com/bookmarklet/ with a different string constant for the column label.
Add this to a bookmark:
```
javascript:(function()%7B(function()%20%7Bvar%20desiredIndex%20%3D%20null%3Bvar%20columnheaders%20%3D%20document.querySelectorAll('%5Brole%3Dcolumnheader%5D')%3Bfor%20(let%20i%20%3D%200%3B%20i%20%3C%20columnheaders.length%3B%20i%2B%2B)%20%7Bif%20(columnheaders%5Bi%5D.firstElementChild.firstChild.innerText%20%3D%3D%20'Estimate%20(d)')%20%7BdesiredIndex%20%3D%20i%3B%7D%7Dvar%20estimateTotal%20%3D%200%3Bvar%20rows%20%3D%20document.querySelectorAll('%5Brole%3Drow%5D')%3Bfor%20(let%20i%20%3D%200%3B%20i%20%3C%20rows.length%3B%20i%2B%2B)%20%7Bif%20(rows%5Bi%5D.children%5BdesiredIndex%5D%20%26%26%20rows%5Bi%5D.children%5BdesiredIndex%5D.firstElementChild%20%26%26%20rows%5Bi%5D.children%5BdesiredIndex%5D.firstElementChild.firstChild)%20%7Bvar%20thing%20%3D%20parseFloat(rows%5Bi%5D.children%5BdesiredIndex%5D.firstElementChild.firstChild.innerText)%3Bif%20(!isNaN(thing))%20%7BestimateTotal%20%3D%20estimateTotal%20%2B%20thing%3B%7D%7D%7Dalert(%22On-screen%20estimate%20total%3A%20%22%20%2B%20estimateTotal%20%2B%20%22%20days%22)%3B%7D)()%7D)()
```
 to your bookmark toolbar and edit the `Estimate (d)` in the URL to your liking. Then just click the bookmarklet on the page, no need for an extension.

This way it works on Firefox as well :)

**Reply by funwithtriangles** [reply]
Thanks for this! I should note that it only works with what's currently being displayed on screen. Any items out of the viewport don't get counted.

**Reply by funwithtriangles** [reply]
One quick way around this is to zoom out really far in your browser

**Comment by funwithtriangles** [comment]
Having the option to see how many story points are assigned to each user in a sprint would also be very useful

**Comment by milanvanschaik** [comment]
Agreed, this would be really helpful to us. One of the few things we're missing while migrating from Jira to Github Projects (Beta). Lovely that the community already has made some extensions/scripts as a temporary fix, but really like to see this natively implemented without a workaround.

**Comment by kimchi-jjigae** [comment]
+1 this is the biggest blocker for my team at the moment!

**Comment by AntieJ** [comment]
+1 Would be very useful for sprint planning

**Comment by nayakravi** [comment]
+1 This will be super useful for planning!

**Comment by kimchi-jjigae** [comment]
Hey all, so at the moment my workaround is to use the "Insights" feature on the project to make a chart, and configure it to sum up the field you want.
Insights is found through the chart icon on the top right of the project, and the URL is at https://github.com/orgs/{org}/projects/{proj_id}/insights/

For example you can make charts like:
<img width="1040" alt="image" src="https://user-images.githubusercontent.com/3331462/167591334-3514189b-93dd-4f4e-94c3-1c90bf127eab.png">

And the Y axis configured like this:
<img width="428" alt="image" src="https://user-images.githubusercontent.com/3331462/167591428-39967992-cc95-497c-8e4b-5131d3f526eb.png">

Hope that helps anyone!

**Reply by jimmycarr** [reply]
Insights is still not available for everyone, sadly. This is a great solution if it is. Does anyone know when insights might move into a wider beta?

**Reply by mochadwi** [reply]
insight exist already in our project, any references to enable the Y-axis options (other than **_Count of items_**)?
<img width="1378" alt="Screen Shot 2022-07-21 at 8 10 02 AM" src="https://user-images.githubusercontent.com/7057959/180108265-a8a5546c-bb6f-4aee-b0bc-c56937d588d9.png">

as we cannot choose it atm

**Reply by henrystaples** [reply]
Add a number field and populate it with data; e.g. we added **Effort** ... but something like **Story Points** would work as well.

![image](https://user-images.githubusercontent.com/29614738/180207742-2a87b175-dda1-473c-aeee-8a6012a38d7e.png)

**Reply by brandonlenz** [reply]
Indeed, just tested this out. The complexity/effort field you add must be a numeric field, so it cannot be a single select, even if all the possible values are numbers (github likely stores them as strings).

It's a slight bummer, because a single select with the fibbonacci values you want to use is convenient for whoever is leading scrum ceremonies. I ended up taking a few minutes to transfer all the values from our single select column to a numeric column because the graphs are much more valuable than the convenience feature of single-select

**Reply by henrystaples** [reply]
right; it would be nice to have the list limited, but it's not a dealbreaker for us. if a team declares 34 hours as the estimate instead of, ohhh, 32 (!), well, so be it. it's just a slight warble which we watch by tracking actuals and variance.

**Comment by Diazbesjorge** [comment]
+1 Insights is nice, but getting totals directly in the board is a must

**Comment by Tenorotti** [comment]
Ideally I would like to be able to add the numbers that I put into custom fields as I'm currently tracking work estimate days so it is kind of useful to have.  Right now, I'm using a calculator or my head.  Any suggestions?

**Comment by ddimatos** [comment]
I would be happy if the next level of progression would be to allow for more than numbers be summed. For example, you might have a size column using the Fibonacci sequence into a single select with clear indications of what those are numerically but you can't sum on those today, i know its not easy to do this for select string type but I can think of ways to do this.

**Comment by jeffpaul** [comment, marked as answer]
Looks like this is [now available](https://github.blog/changelog/2022-09-27-the-new-github-issues-september-27th-update/), yeah?
> ➕ Display sum of numeric fields
> 
> Addressing a [top user request](https://github.com/community/community/discussions/5376), you can now display the sum of a numeric field on the group header in both the table and board layouts.
> 
> Select one or multiple numeric fields you would like to display the total for from the view configuration menu to assist with your planning! 📝

**Comment by alleyhector** [comment]
I just want to echo that allowing a single select field to be summable is what I was searching for when I came across this discussion.

Currently we made the opposite decision as henrystaples because it was more important to us to use fibonacci as well as having a visual representation of the level of effort. We've used animal emojis that represent small to large creatures but this is more than just cute, it actually gives a good analogy when especially newer devs are trying to estimate a task. 

That said, I would really love to be able to tally these in charts and make sure everyone has certain point targets etc.

**Comment by brettwillis** [comment]
As [mentioned](https://github.blog/changelog/2022-09-27-the-new-github-issues-september-27th-update/), we now have a basic version of this feature available, but I'd like to raise a couple of limitations
1. It seems to struggle with number precision (in the example below, the summed numbers are only one decimal place, so I don't know where the other decimal places come from) <br> <img width="307" alt="Screenshot 2022-10-09 at 12 01 04 PM" src="https://user-images.githubusercontent.com/6925974/194730601-eca526c5-9e89-4df2-8899-78f5e0e03647.png">
2. It is only available on the group headers when field grouping is enabled. It would be great to be able to show totals of the entire table (with or without grouping)

**Reply by pete-hotchkiss** [reply]
Have to say - I find it astonishing that 3 years later such a basic - and highly requested - feature still doesn't exist. 

Reading the development blog seems GH are all in on LLM and AI, so perhaps somebody could 'vibe' code the feature.

**Comment by hungluu** [comment]
Upppppp, still waiting for years until I can completely migrate every planning files / tools to github project 👍🏻. Some of my usual use cases:

- Sum of a range of cells or just some cell, example formula: SUM(B1:G1)
- Reference of another cell, for e.g actual completion date of phase 1 is C3, so start date of phase 2 is D3 =C3 + 1d
- Selecting some number cells show their SUM on header or footer bar
